users/models.py

```diff
@@ -291,6 +291,20 @@ def github_profile_url(self):
             return None
         return f"https://github.com/{self.github_username}"
 
+    @staticmethod
+    def get_user_by_github_url(url: str):
+        if not url:
+            return None
+        github_user = url.split("/")[-1]
+
+        if not github_user:
+            return None
+        try:
+            user = User.objects.get(github_username=github_user)
+        except User.DoesNotExist:
+            user = None
+        return user
```

Comment:
Simpler? Maybe we should add `unique=True` to the `github_username` field, though. Seems like a good idea anyway.

```suggestion
        return User.objects.filter(github_username=github_user).first()
```

Reply:
I like this better.

Re: `unique=True` I'm wary of there being users with multiple accounts related to the same github user (e.g. Ion has two accounts, an @example.com account and an @gmail.com account iirc) - I don't have the live data set up yet to confirm one way or the other if that would cause problems so I won't make that change just yet until I have that done.

Given this is just for reports for now and the names are set on users it won't be an issue if it pulls the incorrect user record anyway, but I'll add a todo for it.